homeassistant/components/mqtt/__init__.py

```diff
@@ -439,7 +441,9 @@ async def _reload_config(call: ServiceCall) -> None:
 
         async_register_admin_service(hass, DOMAIN, SERVICE_RELOAD, _reload_config)
 
-    async def async_forward_entry_setup_and_setup_discovery(config_entry):
+    async def async_forward_entry_setup_and_setup_discovery(
+        config_entry: ConfigEntry,
```

Comment:
I suggest that pass the config as an argument to the method, then you don't need to assert anymore.
```suggestion
        config_entry: ConfigEntry, conf: ConfigType,
```